assets/js/Ioda/pages/dashboard/Dashboard.js

```diff
@@ -1,229 +1,227 @@
+  // Tabs
+  const [activeTabType, setActiveTabType] = useState(entityType);
+  //Tab View Changer Button
+  const [tabCurrentView, setTabCurrentView] =
+      useState(entityType === asn.type ? TAB_VIEW_TIME_SERIES : TAB_VIEW_MAP);
+
+  // Map Data
+  const [topoData, setTopoData] = useState(null);
+  const [topoScores, setTopoScores] = useState(null);
+  // Summary Table
+  const [summaryDataRaw, setSummaryDataRaw] = useState(null);
+  const [summaryDataProcessed, setSummaryDataProcessed] = useState([]);
+  // Determine when data is available for table so multiple calls to populate the table aren't made
+  const [genSummaryTableDataProcessed, setGenSummaryTableDataProcessed] = useState(false);
+  const [totalOutagesState, setTotalOutagesState] = useState(0);
+
+  // Summary Table Pagination
+  const [apiPageNumber, setApiPageNumber] = useState(0);
+  // Event Data for Time Series
+  const [eventDataRaw, setEventDataRaw] = useState([]);
+  const [eventDataProcessed, setEventDataProcessed] = useState([]);
+  const [eventOrderByAttr, setEventOrderByAttr] = useState("score");
+  const [eventOrderByOrder, setEventOrderByOrder] = useState("desc");
+  const [eventEndpointCalled, setEventEndpointCalled] = useState(false);
+  const [totalEventCount, setTotalEventCount] = useState(0);
+
+  const [displayDashboardTimeRangeError, setDisplayDashboardTimeRangeError] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+
+
+  useEffect(() => {
+    const timeDiff = until - from;
     if (timeDiff <= 0) {
-      this.setState({
-        displayDashboardTimeRangeError: true,
-      });
+      setDisplayDashboardTimeRangeError(true);
     } else if (timeDiff < dashboardTimeRangeLimit) {
-      this.setState({ mounted: true }, () => {
-        // Set initial tab to load
-        this.handleSelectTab(this.tabs[this.state.activeTabType]);
-        // Get topo and outage data to populate map and table
-        if (this.state.activeTabType !== asn.type) {
-          this.getDataTopo(this.state.activeTabType);
-        }
-        this.getDataOutageSummary(this.state.activeTabType);
-        this.getTotalOutages(this.state.activeTabType);
-      });
+      setMounted(true);
+    }
+  }, [until, from]);
+
+  useEffect(() => {
+    if(!mounted) return;
+    if(activeTabType){
+      handleSelectTab(tabs[activeTabType]);
+      if (activeTabType !== asn.type) {
+        getDataTopo(activeTabType);
+      }
+      getDataOutageSummary(activeTabType);
+      getTotalOutages(activeTabType);
     }
-  }
 
-  componentWillUnmount() {
-    this.setState({
-      mounted: false,
-    });
-  }
+  }, [mounted, activeTabType, asn.type]);
 
-  componentDidUpdate(prevProps, prevState) {
-    // A check to prevent repetitive selection of the same tab
-    if (this.props.tab !== prevProps.tab) {
-      this.handleSelectTab(this.tabs[prevProps.tab]);
+  useEffect(() => {
+    if (tab) {
+      handleSelectTab(tabs[tab]);
     }
+  }, [tab]);
```

Comment:
Don't think we need an effect for this. The handler function should be able to check if the incoming tab is different from the one already set in state.